### PR TITLE
Replace GtkSpinner (#1204242)

### PR DESCRIPTION
**Pull request by dashea**
Evolution had the same problem, so steal their fix.

The spinning image comes out a good bit lighter than GtkSpinner. This may have something to do with how gtk processes symbolic icons? I don't know. I put the steps I took to create the png into a comment in the widget code. Something weird and dumb is going on with what gtk thinks a color is and if anyone can fix that that would be great.

### CODE REVIEW

**Comment by dashea** [comment]
In case anyone tries to compare Spinner.c with the code in Evolution, it's not quite the simple search/replace that I claimed. Setting the "active" property at construct time didn't work because it started the animation before the widget was shown, or something along those lines, so I added the realize/unrealize bits. I will forward those changes back to Evolution.

File: pyanaconda/ui/gui/spokes/lib/refresh.glade [hunk 1]
```diff
@@ -265,9 +266,9 @@ installation options while this scan completes.</property>
                       <object class="GtkLabel" id="label12">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
-                        <property name="xalign">0</property>
                         <property name="label" translatable="yes">Pressing 'OK' below will take you to the disk selection screen where you will need to re-select your disks.</property>
                         <property name="wrap">True</property>
+                        <property name="xalign">0</property>
```
Comment (vpodzime) [inline pyanaconda/ui/gui/spokes/lib/refresh.glade]: Looking at this change and many of the above changes -- would you mind separating the two steps of "*save everything with new glade*" and "*do the changes*" into two separate commits?

File: widgets/glade/AnacondaWidgets.xml [hunk 2]
```diff
@@ -142,14 +142,34 @@
...
-      </properties>
-	</glade-widget-class>
+        <glade-widget-class title="Layout Indicator"
+                            name="AnacondaLayoutIndicator"
+                            icon-name="widget-gtk-label"
+                            generic-name="AnacondaLayoutIndicator">
+            <properties>
+                <property id="label-width" default="12" />
+            </properties>
+        </glade-widget-class>
```
Comment (vpodzime) [inline widgets/glade/AnacondaWidgets.xml]: Same here, I think this change should not be part of the commit or should be mentioned in the commit message.

File: widgets/glade/AnacondaWidgets.xml [hunk 3]
```diff
@@ -142,14 +142,34 @@
...
+
+        <glade-widget-class title="Spinner"
+                            name="AnacondaSpinner"
+                            icon-name="widget-gtk-spinner"
+                            generic-name="AnacondaSpinner"
+                            since="3.2">
+            <properties>
+                <!-- Remove properties inherited from GtkImage -->
+                <!-- Setting disabled doesn't hide them in the glade interface
+                     (nor does setting visible, oddly), but it keeps them being
```
Comment (vpodzime) [inline widgets/glade/AnacondaWidgets.xml]: "...keeps them **from** being saved"?

File: widgets/src/Spinner.c [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+}
+
+static void
+anaconda_spinner_constructed (GObject *object)
+{
+	AnacondaSpinner *spinner;
+	GdkPixbuf *main_pixbuf;
+	gint xx, yy, width, height;
+	GError *error = NULL;
+        gchar *filename;
```
Comment (vpodzime) [inline widgets/src/Spinner.c]: extra space at the beginning of the line

File: widgets/src/Spinner.c [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+
+	/* Chain up to parent's method. */
+	G_OBJECT_CLASS (anaconda_spinner_parent_class)->constructed (object);
+
+	spinner = ANACONDA_SPINNER (object);
+
+        filename = g_strdup_printf("%s/pixmaps/%s", anaconda_get_widgets_datadir(), MAIN_IMAGE_FILENAME);
+        main_pixbuf = gdk_pixbuf_new_from_file (filename, &error);
+        g_free(filename);
```
Comment (vpodzime) [inline widgets/src/Spinner.c]: same here

File: widgets/src/Spinner.c [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+}
+
+static void
+anaconda_spinner_finalize (GObject *object)
+{
+	AnacondaSpinner *spinner = ANACONDA_SPINNER (object);
+
+	g_slist_free_full (spinner->priv->pixbufs, g_object_unref);
+	spinner->priv->pixbufs = NULL;
+	spinner->priv->current_frame = NULL;
```
Comment (vpodzime) [inline widgets/src/Spinner.c]: If I remember that correctly, ``finalize`` is class' finalization and ``dispose`` is instance's finalization so the three lines above should go to ``dispose`` as these resources are per-instance.

**Comment by vpodzime** [comment]
Apart from the comments above this looks good to me. I'm just hesitating to give it an ACK because we are (once again) working around a serious bug (and regression) in Gtk. And the fact that we were able to take the code from **Evolution** which is part of the GNOME project only makes me feel more sad about this. How many other tools/applications are going to copy&paste&modify this code to work around the bug/regression? I don't like doing such hacks just because somebody else is reluctant to fix their bug even if it's that serious regression.Thus not an ACK from me at this point.

File: widgets/glade/AnacondaWidgets.xml (hunk 3 above)
Comment (dashea) [inline widgets/glade/AnacondaWidgets.xml]: Yes. The spinner derives from GtkImage, but none of the GtkImage properties should be applied to the spinner.
